src/station.cpp

```diff
@@ -713,6 +714,44 @@
 	return total_cost >> 3;
 }
 
+/**
+ * Create a hangar on the airport.
+ */
+void Airport::AddHangar()
+{
+	assert(this->hangar == nullptr);
+	assert(Depot::CanAllocateItem());
+	assert(this->GetNumHangars() > 0);
+	Station *st = Station::GetByTile(this->GetHangarTile(0));
+	this->hangar = new Depot(this->GetHangarTile(0));
```

Comment:
> Sorry, but I couldn't comment it above.
> Removal of hangar-depots is handled in RemoveAirport. While CleaningPools, nothing is to be done.
> 
> I don't think it should be handled on ~Station(), as it happens some time after RemoveAirport is called, so it has no airport nor hangar. I don't know how to silence the warning, though.
> 
> I have also checked how are Depots handled in takeovers and bankruptcy of companies. On takeovers, no change is needed. On bankruptcy and complete removal of the company items, RemoveAirport is called, so it is already handled.

Ah, and `delete this->hangar;` shouldn't be in the destructor because the airport can be rebuilt to add or remove a hanger without the airport itself being deleted.

I can silence the warning.

Reply:
Actually, can it be moved to the destructor?

I am not fully clear what persists after a station is demolished to allow a new station to take the name (and orders) of a recently-demolished station. Perhaps you can investigate. 🙂

Reply:
I think it shouldn't be moved to the destructor.

If the airport is deleted, the hangar is immediately deleted as well, **but the station will still exist for some time**, and the station will keep its name until it is destroyed (or the station could have other facilities so the station won't be removed at all).

There would be other ways to deal with the hangars: we could create a depot for each station, independently of whether the station has an airport, so no need to worry to remove the hangar. But I don't like it: creating or removing the hangars when needed seems like a better approach.

Reply:
Fair enough. I will silence the warning.